dev/io.openliberty.jaxws_fat/test-applications/saajclient/src/fats/cxf/jaxws22/saaj/client/SAAJTestServlet.java

```diff
@@ -0,0 +1,89 @@
+import javax.xml.soap.MessageFactory;
+import javax.xml.soap.MimeHeaders;
+import javax.xml.soap.SOAPBody;
+import javax.xml.soap.SOAPConnection;
+import javax.xml.soap.SOAPConnectionFactory;
+import javax.xml.soap.SOAPConstants;
+import javax.xml.soap.SOAPElement;
+import javax.xml.soap.SOAPEnvelope;
+import javax.xml.soap.SOAPException;
+import javax.xml.soap.SOAPFactory;
+import javax.xml.soap.SOAPMessage;
+import javax.xml.soap.SOAPPart;
+
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+
+@SuppressWarnings("serial")
+@WebServlet(urlPatterns = "/SAAJTestServlet")
+public class SAAJTestServlet extends FATServlet {
+
+    /**
+     * Test Description:
+     * Test to make sure that a SAAJ meta-factory successfully created
+     * and we don't have duplicated namespace caused by a bug located in SAAJ
+     * 
+     * Expected Result:
+     * Service should be created and an echo response given as a reply.
+     *
+     */
+    @Test
+    public void testSAAJ() throws Exception {
+        MessageFactory messageFactory = MessageFactory.newInstance();
+        SOAPMessage soapMessage = messageFactory.createMessage();
+        SOAPPart soapPart = soapMessage.getSOAPPart();
+
+        // Set the SOAP version
+        SOAPEnvelope envelope = soapPart.getEnvelope();
+        envelope.addNamespaceDeclaration("s", "http://www.w3.org/2003/05/soap-envelope");
+        envelope.addNamespaceDeclaration("t", "http://www.example.org/types");
+
+        // Create a SOAP body
+        SOAPBody soapBody = envelope.getBody();
+        SOAPElement soapBodyElement = soapBody.addChildElement("echo", "t", "http://www.example.org/types");
+        SOAPElement soapBodyText = soapBodyElement.addChildElement("text", "t", "http://www.example.org/types");
+        soapBodyText.addTextNode("Hello, world!");
+
+        SOAPFactory SOAP_FACTORY = null;
+        // Check if SAAJ meta-factory is successfully created
+        try {
+            SOAP_FACTORY = SOAPFactory.newInstance(SOAPConstants.SOAP_1_1_PROTOCOL);
+        } catch (SOAPException se) {
+            assertFalse(se.getMessage(), se.getMessage().startsWith("Unable to create SAAJ meta-factory: Provider"));
+            return;
+        }
+        SOAPElement getEchoElement = SOAP_FACTORY.createElement("getEcho", "ns0", "http://objects.get.echo.io");
+        SOAPElement echoElement = SOAP_FACTORY.createElement("echo", null, "");
+        getEchoElement.addChildElement(echoElement);
+        soapBody.addChildElement(getEchoElement);
+
+        ByteArrayOutputStream byteOut = new ByteArrayOutputStream();
+        soapMessage.writeTo(byteOut);
+        String soapMessageString = new String(byteOut.toByteArray());
+        
+        // Check if duplicated namespace SAAJ bug is recessed
+        assertFalse("Duplicated namespace in SOAP Body!", soapMessageString.contains("<ns0:getEcho xmlns=\"http://objects.get.echo.io\" xmlns:ns0=\"http://objects.get.echo.io\">"));
```

Comment:
We've had a lot of test failures around how XML is serialized, we need to change the assert to something that doesn't rely on namespace ordering an specific prefixes.